File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    p = {}
    for i in participant:
        if i not in p:
            p[i] = 1
        else:
            p[i] += 1
    for i in completion:
        p[i] -= 1
    for i in participant:
        if p[i] ==  1:
            return i

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
// 해시는 거의 객체 사용

function solution(participant, completion) {
    
    let players = {}
    let answer = ""
    
    // 참여자를 객체에 추가
    participant.forEach((name) => {
        if(players[name]==undefined) players[name] = 1
        else players[name] += 1
    });
    
    // 완주자 객체에서 제거
    completion.forEach((name) => {
        players[name] -= 1
    })
    
    // 객체에 남은 플레이어 중 0이 아닌 플레이어 추출
    players = Object.entries(players).filter(info=>info[1]!=0).map(info=>info[0])    
    answer = players.join("")
    return answer
}